workflowsteps.md

```diff
@@ -0,0 +1,31 @@
+# Workflow Steps Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+A workflow step represents a single step in the reviewing process, e.g. the accept/reject step.  
+
+All endpoints mentioned here require authentication, but no specific permissions.
+
+## Main Endpoint
+**/api/config/workflowsteps**   
+
+Provide access to the configured workflow steps. It returns the list of configured workflow-steps.
+
+## Single Workflow Step Definition
+**/api/config/workflowsteps/<:step-name>**
+
+Provide detailed information about a specific workflow step. An example JSON response document:
+```json
+{
+  	"id": "editstep",
+  	"actions": [
+  	    "approve",
+  	    "reject",
+  	    "edit_metadata"
```

Comment:
I'm curious where this list of actions will come from in the implementation?  It doesn't seem to match the `<action id="">` values in the default [`workflow.xml`](https://github.com/DSpace/DSpace/blob/master/dspace/config/workflow.xml) or those defined in [`workflow-actions.xml`](https://github.com/DSpace/DSpace/blob/master/dspace/config/spring/api/workflow-actions.xml)  Is this list based on those backend configurations?  Or is this defined elsewhere?

Reply:
In DSpace 6, this is defined in [XMLUI AcceptEditRejectAction](https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/xmlworkflow/actions/processingaction/AcceptEditRejectAction.java#L113). The actions in [`workflow.xml`](https://github.com/DSpace/DSpace/blob/master/dspace/config/workflow.xml) don't hold sufficient information.

In DSpace 7, the [API AcceptEditRejectAction](https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/xmlworkflow/state/actions/processingaction/AcceptEditRejectAction.java) still exists to process the data, and is the only code that's aware of the potential action

I was planning to extend the [Action](https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/xmlworkflow/state/actions/Action.java) with a method to request the actions, which is implemented in e.g. the [API AcceptEditRejectAction](https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/xmlworkflow/state/actions/processingaction/AcceptEditRejectAction.java)

The actions will indeed trigger the POST from https://github.com/DSpace/Rest7Contract/blob/master/claimedtasks.md#post-method-single-resource-level, I've included a link to that part. I only wasn't sure whether it's useful to add a "submit_" prefix for each action.

Reply:
@benbosman : OK, I see how you plan to have this work now.  That makes sense. 

One note here though, is that the multiple usages of the term "action" is quite confusing here.  In this example, you are using `action` to refer to what amounts to a UI button press, or a sub-action of the `editaction` (which is defined by `AcceptEditRejectAction`) [configured in the `editstep`](https://github.com/DSpace/DSpace/blob/master/dspace/config/workflow.xml#L26-L33). 

So, when I reviewed this initially, I thought the `actions` here corresponded to `Action` classes....which they do *not*.  Instead, these `actions` are almost like "sub-actions" that can only be performed in the `AcceptEditRejectAction` (used by the `editstep`).  Am I understanding that correctly?

If my understanding is correct, I wonder if we should rename this `action` section to use a different (but similar) term to avoid this confusion?  We could call these `options` or something similar, as these are the options available within this specific Workflow step.  Additionally, the term `options` is also used in the REST Contract here (where these are called `command options`): https://github.com/DSpace/Rest7Contract/blob/master/claimedtasks.md#post-method-single-resource-level

Reply:
Thanks @tdonohue I can understand the terminology confusion, and options is definitely a better name

The claimed task id is actually an id in the database for the specific item, while this endpoint details the specific options that are available. The options are not specific to the item

Reply:
Thanks for clarifying about the `id` column. That makes sense.

I think this Contract is likely "good enough" for now.  That said, as we get into implementation, we may or may not find it makes more sense to eventually align it more with the `/claimedtasks` endpoint.  For example, as you noted, we might eventually want to decide whether these `options` should really be the `parameters` and prefix them with `submit_*` (like in the `claimedtasks` endpoint: https://github.com/DSpace/Rest7Contract/blob/master/claimedtasks.md#post-method-single-resource-level).  

Additionally, we may or may not want to make it clearer the differences between *steps* and *actions*.  This endpoint uses `steps` (which I think is appropriate), while `claimedtasks` talks more about "workflow actions".  This might eventually get confusing to developers, as it's not exactly clear in our REST Contract that the `editstep` Step (in `/workflowsteps`) corresponds to the `editaction` Action (in `/claimedtasks`).  I'm not entirely sure how best to clarify this, but I'm just pointing out the differences in language here.

In any case, I think this is a good first step, so I'll approve this as-is.  Additional updates can always come in future PRs if we deem them necessary.